packages/ogmios/src/ogmiosToCore/tx.ts

```diff
@@ -65,319 +53,305 @@ const mapPoolParameters = (poolParameters: Schema.PoolParameters): Cardano.PoolP
-  if (isNativeScript(script)) {
-    return nativeScript(script.native);
-  } else if (isPlutusV1Script(script) || isPlutusV2Script(script)) return mapPlutusScript(script);
-
-  throw new SerializationError(SerializationFailure.InvalidScriptType, `Script '${script}' is not supported.`);
+  switch (script.language) {
+    case 'native':
+      return nativeScript(script.json);
+    case 'plutus:v1':
+    case 'plutus:v2':
+      return {
+        __type: Cardano.ScriptType.Plutus,
+        bytes: HexBlob(script.cbor),
+        version: script.language === 'plutus:v1' ? Cardano.PlutusLanguageVersion.V1 : Cardano.PlutusLanguageVersion.V2
+      };
+    default:
+      throw new SerializationError(SerializationFailure.InvalidScriptType, `Script '${script}' is not supported.`);
+  }
 };
 
-const mapBootstrapWitness = (b: Schema.BootstrapWitness): Cardano.BootstrapWitness => ({
+const mapBootstrapWitness = (b: Schema.Signatory): Cardano.BootstrapWitness => ({
   // Based on the Ogmios maintainer answer  https://github.com/CardanoSolutions/ogmios/discussions/285#discussioncomment-4271726
   addressAttributes: b.addressAttributes ? Base64Blob(b.addressAttributes) : undefined,
   chainCode: b.chainCode ? HexBlob(b.chainCode) : undefined,
-  key: Crypto.Ed25519PublicKeyHex(b.key!),
-  signature: Crypto.Ed25519SignatureHex(HexBlob.fromBase64(b.signature!))
+  key: Crypto.Ed25519PublicKeyHex(b.key),
+  signature: Crypto.Ed25519SignatureHex(HexBlob.fromBase64(b.signature))
 });
 
 const mapRedeemer = (key: string, redeemer: Schema.Redeemer): Cardano.Redeemer => {
   const purposeAndIndex = key.split(':');
+  const { memory, cpu: steps } = redeemer.executionUnits;
 
   return {
     data: Serialization.PlutusData.fromCbor(HexBlob(redeemer.redeemer)).toCore(),
-    executionUnits: redeemer.executionUnits,
+    executionUnits: { memory, steps },
     index: Number(purposeAndIndex[1]),
     purpose: purposeAndIndex[0] as Cardano.RedeemerPurpose
   };
 };
 
 const mapMetadatum = (obj: Schema.Metadatum): Cardano.Metadatum => {
-  if ('string' in obj) {
-    return obj.string;
-  } else if ('map' in obj) {
-    return new Map(obj.map.map(({ k, v }) => [mapMetadatum(k), mapMetadatum(v)]));
-  } else if ('list' in obj) {
-    return obj.list.map(mapMetadatum);
-  } else if ('int' in obj) {
-    return obj.int;
-  } else if ('bytes' in obj) {
-    return Buffer.from(obj.bytes, 'hex');
+  if (typeof obj === 'string') {
+    return obj;
+  } else if (typeof obj === 'bigint') {
+    return obj;
+  } else if (Array.isArray(obj)) {
+    return obj.map(mapMetadatum);
+  } else if (typeof obj === 'object' && obj !== null) {
+    return new Map(Object.entries(obj).map(([k, v]) => [mapMetadatum(k), mapMetadatum(v)]));
   }
-  throw new NotImplementedError(
-    `Unknown metadatum type: ${typeof obj === 'object' ? Object.keys(obj).join(',') : obj}`
-  );
+  // TODO: deprecate Uint8Array union type from Cardano.Metadatum
```

Comment:
REVIEW: deprecate Uint8Array union type from Cardano.Metadatum

Reply:
Yeah this is odd, worth asking. Also the map is now `{[string]: Metadatum}` which doesn't correspond to cddl model

Reply:
Using CBOR for metadata now